src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -18,37 +22,44 @@
 import org.springframework.web.bind.annotation.RestController;
 
 @RequiredArgsConstructor
+@Validated
 @RestController
 @RequestMapping("api/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of available books")
-    public List<BookDto> getAll(Pageable pageable) {
+    @PreAuthorize("hasRole('ROLE_USER')")
+    public List<BookDto> getAll(@PageableDefault(size = 20, sort = "title",
+            direction = Sort.Direction.ASC) Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
     @Operation(summary = "Get book by ID", description = "Get a book by ID")
+    @PreAuthorize("hasRole('ROLE_USER')")
     public BookDto getBookById(@PathVariable Long id) {
```

Comment:
since you've added `@Validated` - validate here
```suggestion
    public BookDto getBookById(@PathVariable @Positive Long id) {
```

applicable to all places